src/docs/markdown/automatic-https.md

```diff
@@ -220,3 +222,84 @@ Before attempting any ACME transactions, Caddy will test the configured storage
 Caddy can obtain and manage wildcard certificates when it is configured to serve a site with a qualifying wildcard name. A site name qualifies for a wildcard if only its left-most domain label is a wildcard. For example, `*.example.com` qualifies, but these do not: `sub.*.example.com`, `foo*.example.com`, `*bar.example.com`, and `*.*.example.com`.
 
 To get a wildcard from Let's Encrypt, you simply need to enable the [DNS challenge](#dns-challenge) and use a wildcard domain in your config. We recommend using wildcards only when you have so many subdomains that you would encounter CA rate limits trying to obtain certificates for them all.
+
+
+
+## Examples
+
+Caddy implicitly uses the HTTPS port (default 443) for your [site addresses](/docs/conventions#network-addresses) that don't specify a port explicitly (which would disable automatic HTTPS). 
```

Comment:
This is why the docs need improving :+1: 

Although that does make the rule a little confusing. If only an explicit port 80 opts out of HTTPS for a site block, is that really the right way to go about it? (probably too late to change now since v2 is out)

If I add a different port for a service and want to opt-out of HTTPS being implicitly added, how do I do that? Eg SMTP implicit(465) and explicit(587) TLS ports where HTTPS isn't desired?

Reply:
Just tried a config without touching `auto_https`, used `localhost:8000` and connected to it via `curl`, but it responded with:

> Client sent an HTTP request to an HTTPS server.

So HTTPS (or just TLS?) by default. How would I opt out of that for a site block address while other site blocks leverage auto HTTPS? Again important if Caddy is handling the ports for IMAP service that for whatever reason makes plaintext port available to connect to(which can leverage StartTLS, a better example might be the SMTP port 25, which is required to be reachable for incoming mail traffic and is plaintext but can leverage StartTLS to negotiate an explicit TLS connection, something Caddy probably doesn't support and would need to just proxy/passthrough the connection to the SMTP service).

Reply:
> It's fairly nuanced and we need to be careful and rigorous about the language we use, especially in technical documentation.

Agreed. I think that's actually where I got confused about the implicit port changing, because when I first raised an issue about it my global options had `local_certs` set, which I later realized was not required as they're created automatically with `localhost` as the site address.

My comment about explicit port 80 was regarding ports causing an opt-out, but as you've pointed out, that's not correct either.. personally not fond of confusing that can get to initially grok, but I guess for most users it's not a problem they'd need to think about often.

> Caddy's HTTP server is an HTTP server, it doesn't serve SMTP; so ports for other services are irrelevant.
> How (and why) are you doing that with an HTTP server?
> Why are we talking about mail protocols on documentation for an HTTP server?

I'd have services like Postfix and Dovecot handling SMTP/IMAP, they expose the ports. 

Sorry, this was again me not thinking straight. As you rightly point out, we're configuring an HTTP server/service, I have seen a `reverse_proxy` directive but not given it a try yet. I was yet again thinking of alternatives to Caddy that I've used prior where I define a port upfront(which points to the service), and HTTPS especially with the port configurability had me thinking of it as generic automatic TLS with LetsEncrypt support.

I've seen a few users trying to use Caddy with their SMTP services, where Caddy acts as a reverse proxy for those ports, but manages the TLS cert renewal for a hostname (eg mail.example.com). Similar is done with Traefik, while NGINX can actually support handling StartTLS IIRC.

> All the talk about non-HTTP for the "automatic HTTPS" docs really makes no sense, it's totally irrelevant here.

Sorry, as mentioned, I was thinking of it as automatic TLS, and that the only thing that made it HTTPS was the default port and in the context of running an HTTP service.

> Simply break any of the conditions that activate auto-HTTPS? The existing Activation section already answers your question.

It wouldn't break those conditions for `mail.example.com:143`, but as you clarified, that's because this is only related to an HTTP/S service.

> Remember that the HTTP server is only one app module for Caddy, it can definitely have a different module for an SMTP server, with its own semantics.

Understood, I'll need to get more familiar with the [modules](https://caddyserver.com/docs/modules/), I guess anything not in an http namespace might not be supported by Caddyfile and require the JSON approach to configuration(or equivalent via config adapters).